wgengine/netstack/netstack.go

```diff
@@ -704,6 +708,33 @@ func (ns *Impl) injectInbound(p *packet.Parsed, t *tstun.Wrapper) filter.Respons
 	return filter.DropSilently
 }
 
+func (ns *Impl) shouldHandlePing(p *packet.Parsed) (netip.Addr, bool) {
+	if !p.IsEchoRequest() {
+		return netip.Addr{}, false
+	}
+	if !ns.ProcessSubnets {
+		return netip.Addr{}, false
+	}
+
+	destIP := p.Dst.Addr()
+
+	// For non-4via6 addresses, we don't handle pings if they're destined
+	// for a Tailscale IP.
+	if !viaRange.Contains(destIP) {
+		if tsaddr.IsTailscaleIP(destIP) {
+			return netip.Addr{}, false
+		}
+
+		return destIP, true
+	}
+
+	// This is a 4via6 address; translate the destination to an IPv4
+	// address before forwarding it on. Note that despite ICMP4 and ICMP6
```

Comment:
I don't exactly follow this comment but code all looks fine.

Reply:
Expanded/clarified this a bit.